Modules/md5module.c

```diff
@@ -112,6 +112,10 @@ md5_hexdigest(md5object *self)
     unsigned char hexdigest[32];
     int i, j;
 
+    if (PyErr_WarnPy3k_WithFix("md5.hexdigest() returns str in in 3.x",
```

Comment:
If you do `encode(md5.hexdigest(...))` won't you also get this warning?

Reply:
You do actually get the warning, I think I need an extra check.

Reply:
To check if we `hexdigest()` is called inside `encode()` we may have to literally inspect the syntax at grammar level, checking the strings input, I dont seem to see any specific  identifying AST nodes for these. Let me first see if I can extract the statement string from the Function node type before surrendering to the grammar.I am doing some writing today then circle back to this.